### PR TITLE
refactor: squads creation page

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Put the actions on the header even on laptop, then remove the footer.
- Removed a lot of avoidable extra wrappers.
- Cleaned up some unused components after the changes.

Previews:

Screenshot 2024-04-25 at 12 04 04 AM
Screenshot 2024-04-25 at 12 03 59 AM
Screenshot 2024-04-25 at 12 03 55 AM
Screenshot 2024-04-24 at 11 36 03 PM
Screenshot 2024-04-24 at 11 35 59 PM
Screenshot 2024-04-24 at 11 35 50 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-318 #done

### Preview domain
https://mi-318.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/fields/form/FormWrapper.tsx [hunk 1]
```diff
@@ -9,18 +10,22 @@ interface Copy {
 interface ClassName {
   container?: string;
   header?: string;
+  title?: string;
 }
 export interface FormWrapperProps {
   children: ReactNode;
-  className?: string | ClassName;
```
Comment (sshanzel) [inline packages/shared/src/components/fields/form/FormWrapper.tsx:15]: Had to remove the union type. Not really a fan in the first place, but for each and every new property to the `ClassName` object, we will add another line of ternary operator.

File: packages/shared/src/components/squads/Details.tsx [hunk 2]
```diff
@@ -206,150 +164,109 @@ export function SquadDetails({
...
-              label={createMode ? 'Name your Squad' : 'Squad name'}
-              inputId="name"
-              name="name"
-              valid={!!name}
-              leftIcon={<SquadIcon />}
-              value={name ?? ''}
-              className={{
-                container: 'w-full',
-              }}
+      <form
```
Comment (sshanzel) [inline packages/shared/src/components/squads/Details.tsx:167]: Most of the changes here is the wrapper. Maybe I should put them in a separate commit next time.

File: packages/shared/src/components/fields/form/FormWrapper.tsx [hunk 3]
```diff
@@ -29,34 +34,44 @@ export function FormWrapper({
   leftButtonProps = {},
   rightButtonProps = {},
   title,
+  isTitleBesideCancel,
```
Comment (nensidosari) [inline packages/shared/src/components/fields/form/FormWrapper.tsx:37]: is the title beside on placed instead of cancel?

File: packages/shared/src/components/fields/form/FormWrapper.tsx [hunk 4]
```diff
@@ -29,34 +34,44 @@ export function FormWrapper({
...
+          'flex flex-row items-center border-b border-border-subtlest-tertiary px-4 py-2',
+          className?.header,
         )}
         ref={headerRef}
       >
         <Button {...leftButtonProps} variant={ButtonVariant.Tertiary}>
-          {copy?.left ?? 'Cancel'}
+          {left}
         </Button>
+        {isTitleBesideCancel && title && titleElement}
```
Comment (nensidosari) [inline packages/shared/src/components/fields/form/FormWrapper.tsx:64]: Can u attach an image of these 2 variations of title, just not picturing them right in my head

File: packages/shared/src/components/squads/Details.tsx (hunk 2 above)
Comment (nensidosari) [reply packages/shared/src/components/squads/Details.tsx:167]: just to check that this is tested on all screens with all the changes and it still works ok?

**Review by nensidosari — COMMENTED** [review]
Quite a lot of refactoring changes, love that we're removing things, just to make sure we don't miss anything will check the preview domain and see if everything is alright. 
Happy to approve after

**Comment by nensidosari** [comment]
In between some screens the squad details post permissions and below,  info stay centered - maybe we should fix to stay left. Not a big issue though.
Screenshot 2024-04-25 at 10 11 50 AM


Other then that I think it looks all good, tried to check all the affected pages

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/form/FormWrapper.tsx:37]: If we make it part of the button, it looks weird when hovered and it will have a different font size if you are referring to make it as one with the Cancel button.

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/form/FormWrapper.tsx:64]: The other one is the existing layout we have. For example:

Screenshot 2024-04-25 at 6 20 14 PM

**Comment by sshanzel** [comment]
> In between some screens the squad details post permissions and below, info stay centered - maybe we should fix to stay left. Not a big issue though.

They look okay to me, are you maybe looking at the production environment instead of the preview? They kinda change URLs when visiting a Squad.

Screenshot 2024-04-25 at 6 22 21 PM


Screenshot 2024-04-25 at 6 23 08 PM

**Comment by nensidosari** [comment]
> They look okay to me, are you maybe looking at the production environment instead of the preview? They kinda change URLs when visiting a Squad.

Nope I use the url redirector that helps me with this 😄 , did u try it in the screen size I had above 646px...something like this, it is just in specific part, so that's why I said not much important anyways

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 3 above)
Comment (nensidosari) [reply packages/shared/src/components/fields/form/FormWrapper.tsx:37]: No not what I ment, what I'm seeing is, we either show 
- left arrow
- left arrow and title
- cancel button

so `isTitleBesideCancel` naming think gives another sense as we show cancel button and title, or do we call the back arrow cancel?

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 4 above)
Comment (nensidosari) [reply packages/shared/src/components/fields/form/FormWrapper.tsx:64]: Cool thank u, commented above about what I ment https://github.com/dailydotdev/apps/pull/3043#discussion_r1579357147

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/form/FormWrapper.tsx:37]: Right good point. It would probably be weird if we have a "Cancel" and a title. But yes, we do call the back arrow as "Cancel". I will update it to make this more straightforward.

File: packages/shared/src/components/squads/Details.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/Details.tsx:167]: Yep. The screenshots I attached were taken after the changes.

**Comment by sshanzel** [comment]
> Nope I use the url redirector that helps me with this 😄 , did u try it in the screen size I had above 646px...something like this, it is just in specific part, so that's why I said not much important anyways

I see. After checking the config, it seems that falls under mobile now, not sure if we should still consume full-width on that size.

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/form/FormWrapper.tsx]: Let me know if the changes here make it a bit more understandable https://github.com/dailydotdev/apps/pull/3043/commits/a34311396b8460a0c369c170bb9efb6140bdeeb7

Comment (nensidosari) [reply packages/shared/src/components/fields/form/FormWrapper.tsx]: I think still the naming is a bit confusing, since its more related to title, what about `hasHeaderTitle` or `isHeaderTitle`?

**Review by nensidosari — APPROVED** [review]
Looks good, left one last comment on the title thing naming but non blocking

File: packages/shared/src/components/fields/form/FormWrapper.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/form/FormWrapper.tsx]: Updated. It is indeed much clearer now with `isHeaderTitle`.
